src/context.rs

```diff
@@ -192,15 +195,41 @@ pub struct Context<'a, 'b: 'a, 's> {
 
 ///A reader for a request body.
 pub struct BodyReader<'a, 'b: 'a> {
-    request: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>
+    reader: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>,
+    is_multipart: bool,
```

Comment:
Sounds completely reasonable, now that I think about it. I guess asking for the boundary would be enough, then.

Anyway, I'll change this as soon as I get some more free time. I'm still quite occupied...